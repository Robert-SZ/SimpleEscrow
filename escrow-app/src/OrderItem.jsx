/**
 * Created by RobertSabiryanov on 16.08.17.
 */
import React from 'react';
import PropTypes from 'prop-types';

class OrderItem extends React.Component {



    render() {
        const {item}=this.props;
        return (<tr>
            <td>{item.title}</td>
            <td>{item.usedPercentage+'/'+item.amount}</td>
            <td>{item.paticipantsCount}</td>
            <td>{item.usedPercentage<100?<a href="#" onClick={this.props.join}>Join</a>:null}</td>
        </tr>)
    }
}

OrderItem.propTypes={
    join: PropTypes.func.isRequired
};

export default OrderItem