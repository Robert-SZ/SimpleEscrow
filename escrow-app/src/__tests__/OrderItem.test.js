import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import OrderItem from '../OrderItem'

describe('App Tests', () => {
    test('Show request item', () => {
        let item={
            id: 1,
            title: 'Test Title',
            amount: 100,
            usedPercentage: 0,
            paticipantsCount: 0
        };
        const tree = renderer.create(<OrderItem item={item} join={jest.fn}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('Show join button', ()=>{
        let item={
            id: 1,
            title: 'Test Title',
            amount: 100,
            usedPercentage: 0,
            paticipantsCount: 1
        };
        const renderedComponent = shallow(<OrderItem item={item} join={jest.fn}/>);
        expect(renderedComponent.find('a').length).toBe(1);
    })
    test('Hide join button', ()=>{
        let item={
            id: 1,
            title: 'Test Title',
            amount: 100,
            usedPercentage: 100,
            paticipantsCount: 1
        };
        const renderedComponent = shallow(<OrderItem item={item} join={jest.fn}/>);
        expect(renderedComponent.find('a').length).toBe(0);
    })

});