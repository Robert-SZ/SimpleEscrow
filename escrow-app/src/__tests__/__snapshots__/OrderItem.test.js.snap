// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Tests Show request item 1`] = `
<tr>
  <td>
    Test Title
  </td>
  <td>
    0.00/100
    Â ETH
  </td>
  <td>
    0
  </td>
  <td>
    <a
      href="#"
      onClick={[Function]}
    >
      Join
    </a>
  </td>
</tr>
`;
